lib/ecto/query.ex

```diff
@@ -961,6 +966,8 @@ defmodule Ecto.Query do
         end
       end
 
+    {t, quoted} = maybe_with_ties(kw, quoted, binds)
```

Comment:
```suggestion
    {t, quoted} = maybe_with_ties(type, t, quoted, binds)
```

So we don't match on limit again. :)